### PR TITLE
Search - Blue border on the group expense is cut off when the expense is selected

### PR DESCRIPTION
**Pull request by shahinyan11**
### Fixed Issues

$ https://github.com/Expensify/App/issues/50316
PROPOSAL: https://github.com/Expensify/App/issues/50316#issuecomment-2396549091

### Tests

1. Go to staging.new.expensify.com
2. Submit two expenses to any user.
3. Go to Search > Outstanding.
4. Use keyboard to navigate to the group expense in Step 2.
5. Select an expense via checkbox.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as in the Tests section.
### QA Steps

Same as in the Tests section.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (Reported [here](https://github.com/Expensify/App/pull/50914#issuecomment-2417131024))
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/00449e5c-4182-4095-adba-140c5654b6be

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/e23f7875-28ad-4fcc-b379-e8b412e4b3f4

</details>

### CODE REVIEW

**Comment by shahinyan11** [comment]
@rayane-djouah There are console errors when going to the search screen, but they are not related to our changes.

File: src/styles/index.ts [hunk 1]
```diff
@@ -5277,6 +5277,10 @@ const styles = (theme: ThemeColors) =>
             borderColor: theme.border,
             padding: 16,
         },
+
+        outsetShadow: {
```
Comment (shawnborton) [inline src/styles/index.ts:5281]: I feel like we could have a better name for this? It is true that it is using a shadow to make the effect happen, but it seems more appropriate to call this something like "selectionOuterBorder" or something? Thoughts?

**Comment by shahinyan11** [comment]
I found a case where the the top border is cut off. It happens only when the item is first in list. 

Screenshot 2024-10-16 at 21 02 09

We can solve this problem by adding `paddingTop: 1` for `contentContainerStyle` of the `SectionList` [here](https://github.com/Expensify/App/blob/cbd4f1749323405462e8cdac33281c81fece39aa/src/components/SelectionList/BaseSelectionList.tsx#L760C33-L760C54). What do you think about it @shawnborton @rayane-djouah

**Comment by shawnborton** [comment]
Hmm I'm not the biggest fan of adding 1px of padding just to account for this.

**Comment by shahinyan11** [comment]
@shawnborton The I can suggest another solution, But it is different from the changes that I added in my [proposal](https://github.com/Expensify/App/issues/50316#issuecomment-2396549091) . 

We can add below style in [here](https://github.com/Expensify/App/blob/0b3569a4f68851c159ecaf06969a79cf73ff4041/web/index.html#L60). This ensures that the container's inner shadow is not obscured by its children, even if the children are 100% wide and tall.

```
        :focus-visible::before {
            content: '';
            position: absolute;
            top: 0;
            left: 0;
            right: 0;
            bottom: 0;
            box-shadow: inset 0px 0px 0px 1px #5AB0FF;
            pointer-events: none;
            z-index: 1;
        }
```

**Comment by shawnborton** [comment]
Can you share a screenshot/video of what that looks like? Thanks!

**Comment by shahinyan11** [comment]
> Can you share a screenshot/video of what that looks like? Thanks!

It works fine in our case but I have tested and found regression in other places

**Comment by shahinyan11** [comment]
@rayane-djouah Yes we can. I agree with you

**Comment by shahinyan11** [comment]
> @shahinyan11 Can we apply [this style](https://github.com/Expensify/App/pull/50914#issuecomment-2419667241) only to the `listItemPressableStyle`?

We can't add this style in `listItemPressableStyle` object. We can add thisnstyle.for specific componemt by using `dataSet` prop

**Comment by shahinyan11** [comment]
@rayane-djouah the implementation provided [here](https://github.com/Expensify/App/pull/50914#issuecomment-2419667241) works onlily for [BaseListItem](https://github.com/Expensify/App/blob/abdfb7233e040a61d1e4433e202a8b644c7455f4/src/components/SelectionList/BaseListItem.tsx#L15) component

**Comment by shahinyan11** [comment]
@rayane-djouah Do you approve of the changes in [this](https://github.com/Expensify/App/pull/50914#issuecomment-2419667241) comment?

**Review by rayane-d — CHANGES_REQUESTED** [review]

**Comment by shahinyan11** [comment]
@rayane-djouah Your suggested variant do not work as expected. But I made other changes to fix square corners

**Review by rayane-d — APPROVED** [review]
LGTM and tests well :+1:

**Comment by shawnborton** [comment]
Screenshots look good from a design perspective 👍

**Review by arosiclair — APPROVED** [review]

**Pull request merged**